src/App.tsx

```diff
@@ -14,6 +19,10 @@ function App() {
     appSessionState,
   });
 
+  useEffect(() => {
+    onStatusUpdate?.(appSessionState.status);
+  }, [appSessionState.status, onStatusUpdate]);
+
```

Comment:
I could add this code in another place, for example in `AppSessionStateContext`, but I think that here is simpler and it is near to when the menu is rendered, so, it ensures that the callback will be called when the new menu content is rendered.